code/Controllers/HomeController.cs

```diff
@@ -42,24 +43,41 @@ public HomeController(BagsApi bagsApi)
 			{
 				var product = await bagsApi.GetProduct(productId ?? 0);
 
-				ViewData["url"] = $"https://bagcupid.com/app/{path.TrimStart('/')}";
+				model.url = $"https://bagcupid.com/app/{path.TrimLeadingSlash()}";
 
-				if (product.Images != null && product.Images.Count() > 0)
-					ViewData["image"] = product.Images
+				if (product?.Images != null && product.Images.Count() > 0)
+					model.image = product.Images
 						.Aggregate((selectedImage, nextImage) => (nextImage.Priority < selectedImage.Priority) ? nextImage : selectedImage)
 						.Large;
 			}
 
-			return View("index");
+			return View("index", model);
 		}
 	}
 
+	public class MetaViewModel
```

Comment:
In case of Multiple View models, We can create Wrapper View Model class which could have MetaViewModel as it's child. This way Meta Tags will always be separated from other view logic